internal/support/scheduler/v2/application/scheduler/queue.go

```diff
@@ -0,0 +1,204 @@
+	m.intervalToExecutorMap[interval.Name] = sc
+	m.executorQueue.Add(sc)
+}
+
+func (m *manager) deleteInterval(sc *Executor) {
+	delete(m.intervalToExecutorMap, sc.Interval.Name)
+	// Mark as Deleted and scheduler will remove it from the queue
+	sc.MarkedDeleted = true
+}
+
+func (m *manager) addIntervalAction(sc *Executor, action models.IntervalAction) {
+	sc.IntervalActionsMap[action.Name] = action
+	m.actionToIntervalMap[action.Name] = sc.Interval.Name
+}
+
+func (m *manager) IntervalByName(intervalName string) (models.Interval, errors.EdgeX) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	sc, exists := m.intervalToExecutorMap[intervalName]
+	if !exists {
+		return models.Interval{},
+			errors.NewCommonEdgeX(errors.KindContractInvalid,
+				fmt.Sprintf("the executor with name %s does not exist", intervalName), nil)
+	}
+	return sc.Interval, nil
+}
+
+func (m *manager) AddInterval(interval models.Interval) errors.EdgeX {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if _, exists := m.intervalToExecutorMap[interval.Name]; exists {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid,
+			fmt.Sprintf("the executor with name : %s already exists", interval.Name), nil)
+	}
+
+	executor := Executor{
+		IntervalActionsMap: make(map[string]models.IntervalAction),
+		MarkedDeleted:      false,
+	}
+	executor.Reset(interval, m.lc)
+	m.addInterval(interval, &executor)
+
+	m.lc.Info(fmt.Sprintf("added the interval with name : %s into the scheduler queue", interval.Name))
+	return nil
+}
+
+func (m *manager) UpdateInterval(interval models.Interval) errors.EdgeX {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	executor, exists := m.intervalToExecutorMap[interval.Name]
+	if !exists {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid,
+			fmt.Sprintf("the executor with name %s does not exist", interval.Name), nil)
+	}
+	executor.Reset(interval, m.lc)
+
+	m.lc.Info(fmt.Sprintf("updated the interval with name: %s in the scheduler queue", interval.Name))
+	return nil
+}
+
+func (m *manager) DeleteIntervalByName(intervalName string) errors.EdgeX {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.lc.Debug(fmt.Sprintf("removing the interval with name: %s ", intervalName))
+
+	sc, exists := m.intervalToExecutorMap[intervalName]
+	if !exists {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid,
+			fmt.Sprintf("the executor with name %s does not exist", intervalName), nil)
+	}
+	m.deleteInterval(sc)
+
+	m.lc.Info(fmt.Sprintf("removed the interval with id: %s from the scheduler queue", intervalName))
+	return nil
+}
+
+func (m *manager) IntervalActionByName(actionName string) (action models.IntervalAction, err errors.EdgeX) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	intervalName, exists := m.actionToIntervalMap[actionName]
+	if !exists {
+		return action,
+			errors.NewCommonEdgeX(errors.KindContractInvalid,
+				fmt.Sprintf("scheduler could not find interval name with interval action name : %s", actionName), nil)
```

Comment:
`interval name` => `interval`

Reply:
Removed to reduntant IntervalActionByName func.